enhancements/cluster-scope-secret-volumes/shared-resource-validation.md

```diff
@@ -0,0 +1,413 @@
+The one time read of the ConfigMap allow list for Shared Resources whose name starts with "openshift-*" is impacted
+in the sense that which Shared Resources the webhook allows could change.  However, the removal of any 4.N+1 Shared Resources
+pre-populated by another OCP controller/operator should be handled by the CVO's management of that operator's manifest
+(i.e. other OCP operators that want to provide pre-populated Shared Resources should do so in their manifest.)
+
+And with duplicating checks in the controller relist or CSI Driver interface with the Kubelet, the desired validations
+will be made even if for some reason and upgrade/downgrade of the webhooks are not started yet, or if the new webhooks are not
+functional for some reason. The only difference is how the prevention of the mount is surfaced, as articulated in the goals of the EP.
+
+### Operational Aspects of API Extensions
+
+Carry over from 4.10 (looking to confirm this with 4.10 metrics): expect a low number (single digits) of SharedSecrets
+and SharedConfigMaps.  Expect consumption of those Shares across many Pods etc. across many namespaces.
+
+Also carry over: RBAC objects that grant permission for Pods' ServiceAccounts in those namespaces to use SharedSecrets / SharedConfigMaps 
+in each of those namespaces are needed.  RBAC so that specific SharedSecrets / SharedConfigMaps can be discovered are also
+most likely.
+
+So the 4.10 impact for Shared Resources on the API Server storage i.e. 'etcd' was deemed "no problem".
+
+#### New SLIs for the new admission validations to help cluster administrators and support
+
+Separate metrics for
+= SharedSecret creation/modification failures because of an openshift-* name.  Metric labels for share name, secret name, secret namespace
+- SharedConfigMap creation/modification failures because of an openshift-* name.  Metric labels for share name, configmap name, configmap namespace
+- SharedConfigMap/SharedSecret having the validity condition go to invalid.  Metric labels for share name, reference name and namespace
+- Pod specable creation/modification failure for Shared Resource CSI Volumes without 'readOnly: true'.  Metric labels for namespace, Group/Kind/Version, share name.
+
+Per open questions: since we are validating at both the webhook and controller (in case webhook is down), do they need separately
+named metrics (i.e. "..._webhook_total" and "..._controller_total") for the same logical items in the above list?
+
+Deferring on any metrics around associated Secret/ConfigMap validation until some of the open questions around that topic
+are answered.
+
+Alert 1:  using the `rate()` prometheus function, we can compare our failure counts above with existing metrics from 4.10
+wrt successful mounts, and build alerts where, for example, if failures / successes over 1 minute > 0.1, fire.  A pseudo-code
+example: `rate(read_only_failures[1m] / rate(successful_mount[1m]) > 0.1`.  Debating info only to moderate priority for the alert
```

Comment:
Is there a general metric for volume mounts succeeding/failing?

Reply:
I have not found one.   I'll look again.  If still no luck, I'll add an open question, reach out to the storage team.

Reply:
I will be posting what I did find as part of an open question on this topic.... summary, "sort of" but it is only alpha level at this time https://kubernetes-csi.github.io/docs/volume-health-monitor.html